### ISSUE TITLE
Fix known issue with Any, Any resolution

### ISSUE THREAD

**Issue by brynrhodes** [cqframework/clinical_quality_language#435]
The following expressions:

    define Test1: null ~ null
    define Test2: null = null

result in an error about ambiguous resolution. This is because the generic resolution that would have enabled these is disabled (see #420) because once a binary operator signature for Any, Any is instantiated, it ends up being used for all resolutions, even if a more appropriate specific resolution is available. The operator resolution needs to take this possibility into account so that this functionality can be reintroduced.

**Comment by cmoesel** [comment]
This is only relevant when both sides are _literal_ `null`, right?  In other words, while this might come up in test cases, you should never see production CQL that does a comparison between two literal `null`s -- so it's really only a problem for tests cases like the one above, right?

**Comment by brynrhodes** [comment]
Yes, it's an edge case for sure, but there are also places where it will manifest if you only have a compile-time type of Any, where an =(Any, Any) should be allowed but the translator can't resolve it.
